### PR TITLE
Fixing flawed bounding box tests and skipped modeling tests

### LINKED ISSUE #12182: Flawed bounding box unit test in modeling
The [`astropy/modeling/tests/test_models.py::TestFittable1DModels::test_bounding_box1D`](https://github.com/astropy/astropy/blob/4f54d9eb59f2937f3470869d377c0c357b96e37f/astropy/modeling/tests/test_models.py#L348-L385) unit test is flawed. In particular, it purposely evaluates the model on values outside the bounding box (see [here](https://github.com/astropy/astropy/blob/4f54d9eb59f2937f3470869d377c0c357b96e37f/astropy/modeling/tests/test_models.py#L380-L381)). This is a huge issue because one useful application of explicitly adding a bounding box to a model class would be to purposely restrict the model's inputs to the mathematical domain for the model.

For example if one were to want to create an `arcsin` based model it is completely reasonable to hard code the bounding box so that the arguments made to `arcsin` are in the interval `[-1, 1]` (otherwise you get a warning if you use `numpy.arcsin` or an error if you use `math.asin`).

It is also reasonable to expect user contributed models to `astropy.modeling.models` to be added to [`astropy/modeling/tests/example_models.py`](https://github.com/astropy/astropy/blob/4f54d9eb59f2937f3470869d377c0c357b96e37f/astropy/modeling/tests/example_models.py) so that it can be tested by the "standard" battery of tests, one of which is the one at issue. So adding a model whose bounding box defines the extent of the true domain of the model will have difficulties passing this test.

Moreover, the [latter section](https://github.com/astropy/astropy/blob/4f54d9eb59f2937f3470869d377c0c357b96e37f/astropy/modeling/tests/test_models.py#L366-L385) of this test seems to be making the incorrect assumption that the bounding box defines the majority of the "support" of the model (e.g. `Gaussian1D`) where the values outside are tiny. This is an incorrect assumption in general so the point of this part of the test is questionable at best.

Finally, this test fails to test using the bounding box in a meaningful way because it does not try to evaluate the model using the `with_bounding_box=True` keyword argument. That enforces the bounding box such that the outputs corresponding to inputs outside the bounding box return NaN values (and are not run through the actual model evaluation machinery). I propose that the part of the test at issue be removed and replaced with test code that tests this precise feature.

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

As pointed out in issue #12182 the bounding box tests being performed were not testing bounding boxes properly. Instead those tests were testing that the "peak" described by the bounding box accounted for > 1% of the integral of the model (checking that integration over the bounding box accounted for most of the area). Conceptually this is not what the bounding box was intended to do, instead it was meant to filter out "bad" input values, e.g. points outside the domain of the model.

This refactors the bounding box test into two tests. One which tests the bounding box as a domain for which values are tested as in or outside, and an additional test which tests the peak property that was originally tested for only for models which where actually being tested for this peak property. This means that adding models with bounding boxes that fail to have the peak property (such as those whose bounding box describes the entire domain of the model, causing the peak property test) will no longer create a mysterious test failure.

Note that this PR also fixes how `astropy/modeling/tests/test_models.py` was handling tests of properties for models that do not have them. Before, the test was being marked as skipped. I do not believe this should be the case the model implicitly should pass those tests rather than listing a test which will always be marked down as skipped.

Finally, while looking at skipped tests throughout `astropy.modeling` I noticed that a polynomial test was being skipped due to a malformed test parameterization, which caused an empty list of test parameters to be passed to the test. I resolved this issue so that the test would become useful once again.

Fixes #12182

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by WilliamJamieson** [comment]
Tagging @perrygreenfield for review.

**Comment by pllim** [comment]
Usually there is no need for change log since tests are not really user-facing code but I'll leave that up to `modeling` maintainers.

I also don't see the need to backport.

**Review by perrygreenfield — APPROVED** [review]
LGTM

**Comment by pllim** [comment]
@perrygreenfield , you think this needs a change log?

**Comment by perrygreenfield** [comment]
For completeness I guess. Most end users won’t care. I’ll leave it for your judgement. If it does need one, let William know.

**Comment by pllim** [comment]
Well, this has a change log but I don't think it needs one... 😬

**Comment by WilliamJamieson** [comment]
> Well, this has a change log but I don't think it needs one... 😬

I have removed the change log. So it should be good to go.

**Pull request merged**
